processing/src/main/java/io/druid/segment/serde/LongGenericColumnPartSerdeV2.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment.serde;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.druid.collections.bitmap.ImmutableBitmap;
+import io.druid.java.util.common.io.smoosh.FileSmoosher;
+import io.druid.segment.column.ValueType;
+import io.druid.segment.data.BitmapSerde;
+import io.druid.segment.data.BitmapSerdeFactory;
+import io.druid.segment.data.CompressedColumnarLongsSupplier;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.nio.ByteOrder;
+import java.nio.channels.WritableByteChannel;
+
+/**
+ */
+public class LongGenericColumnPartSerdeV2 implements ColumnPartSerde
```

Comment:
Double, Float and Long look very similar, could a common base be extracted to reduce the amount of boilerplate?

Reply:
this is out of scope of this PR, we can do this in a separate PR.

Reply:
Please create an issue

Reply:
https://github.com/druid-io/druid/issues/5405